experimental/segmenter/src/complex.rs

```diff
@@ -109,27 +113,42 @@ impl Dictionary {
     /// Construct a [`Dictionary`] for Chinese and Japanese.
     #[cfg(feature = "auto")] // Use by WordSegmenter with "auto" enabled.
     pub(crate) fn new_chinese_japanese<
-        D: DataProvider<UCharDictionaryBreakDataV1Marker> + ?Sized,
+        D: DataProvider<DictionaryForWordOnlyAutoV1Marker> + ?Sized,
     >(
         provider: &D,
     ) -> Self {
-        let cj = Self::load(provider, locale!("ja")).ok();
+        let cj = Self::load_chinese_japanese(provider, locale!("ja")).ok();
         Dictionary {
             cj,
             ..Default::default()
         }
     }
 
+    pub(crate) fn load_chinese_japanese<
```

Comment:
nit: you can keep a single `load` function and make it generic in the marker.

Reply:
Sure; will fix in a follow up